speedy-recovery-client/src/service/FhirMappingAdapterSTU2.js

```diff
@@ -0,0 +1,116 @@
+import {
+  getbodySite,
+  getForm,
+  getLocation,
+  getMedName,
+  getName,
+  getPatient,
+  getPhone,
+  getPractitioner,
+  getSeverity,
+  getSummary
+} from "./FhirJsonCommonExtractionTools";
+
+const mapPatientToUserSTU2 = fhirPatientResource => ({
+  //This is a temporary hard-code fix as we have not implemented the searching for a patients' parent
+  role:
+    fhirPatientResource.id === "f0462936-eb4b-4da1-b45a-fbd96ebf8ccb"
+      ? "Parent"
+      : fhirPatientResource.resourceType,
+
+  name: getName(fhirPatientResource.name),
+  birthDate: fhirPatientResource.birthDate
+    ? new Date(fhirPatientResource.birthDate)
+    : "Unknown",
+  gender: fhirPatientResource.gender ? fhirPatientResource.gender : "Unknown",
+  careProvider: "Undefined in STU2",
+  address: getAddress(fhirPatientResource.address),
+  phone: getPhone(fhirPatientResource.telecom),
+  email: "Undefined in STU2"
+});
+
+const mapAppointmentSTU2 = fhirAppResource => ({
+  id: fhirAppResource.id ? fhirAppResource.id : "Unknown",
+  title: getTitle(fhirAppResource.text),
+  status: fhirAppResource.status ? fhirAppResource.status : "Unknown",
+  appType: "Undefined in STU2",
+  indication: "Undefined in STU2",
+  priority: fhirAppResource.priority ? fhirAppResource.priority : "Unknown",
+  description: fhirAppResource.description
+    ? fhirAppResource.description
+    : "Unknown",
+  supportingInfo: "Undefined in STU2",
+  start: fhirAppResource.start ? new Date(fhirAppResource.start) : "Unknown",
+  end: fhirAppResource.end ? new Date(fhirAppResource.end) : "Unknown",
+  created: "Undefined in STU2",
+  comment: fhirAppResource.comment ? fhirAppResource.comment : "Unknown",
+  patient: getPatient(fhirAppResource.participant),
+  practitioner: getPractitioner(fhirAppResource.participant),
+  location: getLocation(fhirAppResource.participant)
+});
+
+const mapConditionSTU2 = fhirCondResource => ({
+  clinicalStatus: "Undefined in STU2",
+  verificationStatus: fhirCondResource.verificationStatus
+    ? fhirCondResource.verificationStatus
+    : "Unknown",
+  severity: getSeverity(fhirCondResource.severity),
+  summary: getSummary(fhirCondResource.code),
+  bodySite: getbodySite(fhirCondResource.bodySite),
+  onsetDateTime: fhirCondResource.onsetDateTime
+    ? new Date(fhirCondResource.onsetDateTime)
+    : "Unknown"
+});
+
+const mapMedicationSTU2 = fhirMedResource => ({
+  id: fhirMedResource.id ? fhirMedResource.id : "Unknown",
+  producer: "Undefined in STU2",
+  name: getMedName(fhirMedResource.code),
+  isBrand:
+    typeof fhirMedResource.isBrand !== "undefined"
```

Comment:
What's wrong with a simple `fhirMedResource.isBrand !== undefined ? ...`, or even more concise `fhirMedResource.isBrand ? ...` so just check if this field is a truthy value

(same comment applies to similar lines in this file and the other adapter as well)

Reply:
By the way, I often forget how JS works, so what I'll do is just try it in the interactive console in the browser:
![image](https://user-images.githubusercontent.com/11090412/53307647-8177bf00-3892-11e9-933f-3fc200748ed8.png)

Reply:
isBrand has a boolean value. So when i use fhirMedResource.isBrand ? do this : "Unknown"
(if isBrand is present but the value of isBrand is false, the statement returns unknown). But i think we can use the first example.

Reply:
I understand your point now. So I agree that it makes sense to actually test "undefined" rather than any falsy value.

I saw that you are now doing `fhirMedResource.isBrand !== "undefined"`? But this would literally check if the field is present and holding a string value of `"undefined"`. You want to check against `undefined` (no double quotes). This works independently from what data type is in the field. It just checks if the field itself is missing (no value at all)

Reply:
oopss. Looking into it..